src/node/internal/internal_timers_promises.ts

```diff
@@ -0,0 +1,218 @@
+// Copyright (c) 2017-2022 Cloudflare, Inc.
+// Licensed under the Apache 2.0 license found in the LICENSE file or at:
+//     https://opensource.org/licenses/Apache-2.0
+//
+// Adapted from Node.js. Copyright Joyent, Inc. and other Node contributors.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a
+// copy of this software and associated documentation files (the
+// "Software"), to deal in the Software without restriction, including
+// without limitation the rights to use, copy, modify, merge, publish,
+// distribute, sublicense, and/or sell copies of the Software, and to permit
+// persons to whom the Software is furnished to do so, subject to the
+// following conditions:
+//
+// The above copyright notice and this permission notice shall be included
+// in all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS
+// OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+// MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN
+// NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM,
+// DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR
+// OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE
+// USE OR OTHER DEALINGS IN THE SOFTWARE.
+
+import * as timers from 'node-internal:internal_timers';
+import { ERR_INVALID_THIS, AbortError } from 'node-internal:internal_errors';
+import {
+  validateNumber,
+  validateAbortSignal,
+  validateBoolean,
+  validateObject,
+} from 'node-internal:validators';
+
+const kScheduler = Symbol.for('kScheduler');
+
+export async function setTimeout(
+  after: number | undefined,
+  value?: unknown,
+  options: { signal?: AbortSignal; ref?: boolean } = {}
+): Promise<typeof value> {
+  if (after !== undefined) {
+    validateNumber(after, 'delay');
+  }
+
+  validateObject(options, 'options');
+
+  if (options.signal !== undefined) {
+    validateAbortSignal(options.signal, 'options.signal');
+  }
+
+  if (options.ref !== undefined) {
```

Comment:
Add a comment here explaining that `options.ref` is not used but we are going to validate it still for consistency/compat reasons.